src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -268,8 +268,7 @@ private boolean launchNode(SchedulerDriver driver, Offer offer,
     List<TaskInfo> tasks = new ArrayList<>();
     for (String taskType : taskTypes) {
       List<Resource> taskResources = getTaskResources(taskType);
-      String taskName = getNextTaskType(taskType);
-      if (taskName.isEmpty()) return false;
+      String taskName = getNextTaskName(taskType);
```

Comment:
It's a runtime exception so I think should be safe to propagate up. It follows the same pattern as the other exceptions in scheduler.